Tests/Unit/Hook/FileDumpControllerHookTest.php

```diff
@@ -0,0 +1,127 @@
+
+/**
+ * Testcase for FileDumpControllerHook
+ */
+class FileDumpControllerHookTest extends TestCase
+{
+    /**
+     * @var FileDumpControllerHook
+     */
+    protected $fileDumpControllerHook;
+
+    /**
+     * Set up this testcase
+     */
+    protected function setUp()
+    {
+        $this->fileDumpControllerHook = new FileDumpControllerHook();
+    }
+
+    /**
+     * @test
+     *
+     * @dataProvider grantedUserCombinations
+     * @param array $application
+     * @param FrontendBackendUserAuthentication $beUser
+     * @param FrontendUserAuthentication $feUser
+     */
+    public function grantsAccess($application = [], FrontendBackendUserAuthentication $beUser = null, FrontendUserAuthentication $feUser = null)
+    {
+        $this->assertEquals(true, $this->fileDumpControllerHook->hasAccess($application, $feUser, $beUser));
+    }
+
+    /**
+     * @return array
+     */
+    public function grantedUserCombinations()
+    {
+        $feUser = $this->prophesize(FrontendUserAuthentication::class);
+        $feUser->user = [
+            'uid' => '999',
+        ];
+
+        return [
+            'BE logged in, FE not logged in' => [[], $this->prophesize(FrontendBackendUserAuthentication::class)->reveal(), null],
+            'BE not logged in, FE logged in with application connection' => [['user' => '999'], null, $feUser->reveal()],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @dataProvider deniedUserCombinations
+     * @param array $application
+     * @param FrontendBackendUserAuthentication $beUser
+     * @param FrontendUserAuthentication $feUser
+     */
+    public function deniesAccess($application = [], FrontendBackendUserAuthentication $beUser = null, FrontendUserAuthentication $feUser = null)
+    {
+        $this->assertEquals(false, $this->fileDumpControllerHook->hasAccess($application, $feUser, $beUser));
+    }
+
+
+    /**
+     * @return array
+     */
+    public function deniedUserCombinations()
+    {
+        $feUser = $this->prophesize(FrontendUserAuthentication::class);
+        $feUser->user = [
+            'uid' => '999',
+        ];
+
+        return [
+            'BE not logged in, FE not logged in' => [[], null, null],
+            'BE not logged in, FE logged in but no application connection' => [['user' => '123'], null, $feUser->reveal()],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @dataProvider fileLocations
+     * @param  string $filePath The file path to check
+     * @param  bool $returnValue The desired return value
+     */
+    public function checksIfFileIsInAtsStorage($filePath, $returnValue)
+    {
+        $extconfService = $this->prophesize(ExtconfService::class);
+        $extconfService->getUploadConfiguration()->willReturn([
+            'uploadFolder' => '12:/somestorage/',
+        ]);
+
+        GeneralUtility::setSingletonInstance(ExtconfService::class, $extconfService->reveal());
```

Comment:
Please add a `tearDown()` to this testcase with `GeneralUtility::purgeInstances()`.

Reply:
@mbrodala I don't think we need to catch invalid upload Folders here as they are just compared with "startsWith", not converted into actual folder/storage objects. So if someone enters invalid values, nothing will happen, the file protection will simply not work.

Please also see my latest changes, I fixed the raised issues.